docs/secondary-network.md

```diff
@@ -0,0 +1,142 @@
+
+## Prerequisites
+
+Native secondary network support is still an alpha feature and is disabled by
+default. To use the feature, the `SecondaryNetwork` feature gate must be enabled
+in the `antrea-agent` configuration. If you need IPAM for the secondary
+interfaces, you should also enable the `AntreaIPAM` feature gate in both
+`antrea-agent` and `antrea-controller` configuration. At the moment, Antrea IPAM
+is only available IPAM option for secondary networks managed by Antrea. The
+`antrea-config` ConfigMap with the two feature gates enables is like the
+following:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: antrea-config
+  namespace: kube-system
+data:
+  antrea-controller.conf: |
+    featureGates:
+      AntreaIPAM: true
+  antrea-agent.conf: |
+    featureGates:
+      AntreaIPAM: true
+      SecondaryNetwork: true
+```
+
+Antrea leverages the `NetworkAttachmentDefinition` CRD from [Kubernetes Network
+Plumbing Working Group](https://github.com/k8snetworkplumbingwg/multi-net-spec)
+to define secondary networks. You can import the CRD to your cluster using the
+following command:
+
+```bash
+kubectl apply -f https://github.com/k8snetworkplumbingwg/network-attachment-definition-client/raw/master/artifacts/networks-crd.yaml
+```
+
+## Secondary OVS bridge configuration
+
+A VLAN secondary interface will be connected to a separate OVS bridge on the
+Node. You can specify the secondary OVS bridge configuration in the
+`antrea-agent` configuration, and `antrea-agent` will automatically craete the
+OVS bridge on based on the configuration. For example, the following
+configuration will create an OVS bridge named `br-secondary`, with a physical
+interface `eth1`.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: antrea-config
+  namespace: kube-system
+data:
+  antrea-agent.conf: |
+    secondaryNetwork:
+    ovsBridges: [{"bridgeName": "br-secondary", "physicalInterfaces": ["eth1"]}]
+```
+
+At the moment, Antrea supports only a single OVS bridge for secondary networks,
+and supports only a single physical interface on the bridge. The physical
+interface cannot be the Node's management interface, otherwise a Node's
+management network connectivity can be broken after `antrea-agent` creates the
+OVS bridge and moves the management interface to the bridge.
+
+## Secondary VLAN network configuration
+
+A secondary VLAN network is defined by a NetworkAttachmentDefinition CR. For
+example, the following NetworkAttachmentDefinition defines a VLAN network
+`vlan100`.
+
+```yaml
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: vlan100
+spec:
+  config: '{
+      "cniVersion": "0.3.0",
+      "type": "antrea",
+      "networkType": "vlan",
+      "mtu": 1500,
+      "vlan": 100,
+      "ipam": {
+        "type": "antrea",
+        "ippools": ["vlan100-ipv4", "vlan100-ipv6"]
+      }
+    }'
+```
+
+`antrea-agent` will connect a Pod secondary interface of a VLAN network to the
```

Comment:
s/will connect a Pod secondary interface of a VLAN network to the/will connect Pod secondary interfaces belonging to a VLAN network to the

Reply:
Done